packages/db/test/unit/controller/level.test.ts

```diff
@@ -106,4 +113,33 @@ describe("LevelDB controller", () => {
     const result = await all(resultStream);
     expect(result.length).to.be.equal(2);
   });
+
+  it("test compactRange + approximateSize", async () => {
+    const indexes = Array.from({length: 100}, (_, i) => i);
+    const keys = indexes.map((i) => Buffer.from([i]));
+    const values = indexes.map((i) => Buffer.alloc(1000, i));
+    const minKey = Buffer.from([0x00]);
+    const maxKey = Buffer.from([0xff]);
+
+    await db.batchPut(keys.map((key, i) => ({key, value: values[i]})));
+    await db.batchDelete(keys);
+
+    const sizeBeforeCompact = getDbSize();
+    await db.compactRange(minKey, maxKey);
+    const sizeAfterCompact = getDbSize();
+
+    expect(sizeAfterCompact).lt(sizeBeforeCompact, "Expected sizeAfterCompact < sizeBeforeCompact");
```

Comment:
how much disk space was saved with this test? I think we should expect some threshold here

Reply:
I'm not that familiar with level internals so I think it's dangeours to do assumptions here. LevelDB behaviour is not trivial